src/testdrive/src/action/sql.rs

```diff
@@ -183,19 +184,62 @@ impl Action for SqlAction {
             return Err(e);
         }
 
-        if let Some(path) = &state.materialized_data_path {
-            match self.stmt {
-                Statement::CreateDatabase { .. }
-                | Statement::CreateIndex { .. }
-                | Statement::CreateSchema { .. }
-                | Statement::CreateSource { .. }
-                | Statement::CreateTable { .. }
-                | Statement::CreateView { .. }
-                | Statement::DropDatabase { .. }
-                | Statement::DropObjects { .. } => {
+        match self.stmt {
+            Statement::CreateDatabase { .. }
+            | Statement::CreateIndex { .. }
+            | Statement::CreateSchema { .. }
+            | Statement::CreateSource { .. }
+            | Statement::CreateTable { .. }
+            | Statement::CreateView { .. }
+            | Statement::DropDatabase { .. }
+            | Statement::DropObjects { .. } => {
+                let disk_state = if let Some(url) = &state.materialized_catalog_postgres_stash {
+                    let schema = format!("mz_stash_copy_{}", state.seed);
+
+                    let (client, connection) = tokio_postgres::connect(url, NoTls).await?;
+                    mz_ore::task::spawn(|| "tokio-postgres testdrive connection", async move {
+                        if let Err(e) = connection.await {
+                            panic!("postgres stash connection error: {}", e);
+                        }
+                    });
+
+                    client
+                        .execute(format!("CREATE SCHEMA {schema}").as_str(), &[])
+                        .await?;
+                    client
+                        .execute(format!("SET search_path TO {schema}").as_str(), &[])
+                        .await?;
+                    client
+                        .batch_execute(
+                            "
+                                CREATE TABLE fence AS SELECT * FROM public.fence;
+                                CREATE TABLE collections AS SELECT * FROM public.collections;
+                                CREATE TABLE data AS SELECT * FROM public.data;
+                                CREATE TABLE sinces AS SELECT * FROM public.sinces;
+                                CREATE TABLE uppers AS SELECT * FROM public.uppers;
+                            ",
+                        )
+                        .await?;
```

Comment:
Haha, nice. I might drop a comment here that says "TODO: consider adding a read-only API to stash directly" but these seems like exactly the right solution for right now.